Feature - Core Infrastructure

DeVaukz opened the pull request:
- DO NOT MERGE - DO NOT MERGE - DO NOT MERGE *

This is still a WIP.  I'm PR'ing early so everyone can begin complaining about my programming or how terrible of an idea this is in general.  Just kidding, hopefully.

See the ReadMe for details. (I don't intend for the ReadMe to be merged)

adam4813 commented:
As of right now we will focus on using glfw for our platform independence
layer, so the platform\* files are not needed until such a time as we move
to more platform specific code.

Other than that it seems like a good start and we'll have to discuss it in
the meeting.
On Apr 20, 2014 8:03 PM, "Devin" notifications@github.com wrote:

> - DO NOT MERGE - DO NOT MERGE - DO NOT MERGE *
> 
> This is still a WIP. I'm PR'ing early so everyone can begin complaining
> about my programming or how terrible of an idea this is in general. Just
> kidding, hopefully.
> 
> ## See the ReadMe for details. (I don't intend for the ReadMe to be merged)
> 
> You can merge this Pull Request by running
> 
>   git pull https://github.com/DeVaukz/trillek-client-core feature-CoreInfrastructure
> 
> Or view, comment on, or merge it at:
> 
>   https://github.com/trillek-team/trillek-client-core/pull/10
> Commit Summary
> - Begin Core Infrastructure Feature
> 
> File Changes
> - _M_ CMakeLists.txthttps://github.com/trillek-team/trillek-client-core/pull/10/files#diff-0(55)
> - _A_ Engine/CMakeLists.txthttps://github.com/trillek-team/trillek-client-core/pull/10/files#diff-1(20)
> - _A_ Engine/Core/CMakeLists.txthttps://github.com/trillek-team/trillek-client-core/pull/10/files#diff-2(58)
> - _A_ Engine/Core/Core.hhttps://github.com/trillek-team/trillek-client-core/pull/10/files#diff-3(9)
> - _A_ Engine/Core/Misc/Build.hhttps://github.com/trillek-team/trillek-client-core/pull/10/files#diff-4(44)
> - _A_ Engine/Core/Platform/Platform.hhttps://github.com/trillek-team/trillek-client-core/pull/10/files#diff-5(99)
> - _A_ Engine/Core/Platform/PlatformIncludes.hhttps://github.com/trillek-team/trillek-client-core/pull/10/files#diff-6(16)
> - _A_ Engine/Core/PlatformBase/BasePlatform.hhttps://github.com/trillek-team/trillek-client-core/pull/10/files#diff-7(6)
> - _A_ Engine/Core/PlatformMac/MacPlatform.hhttps://github.com/trillek-team/trillek-client-core/pull/10/files#diff-8(35)
> - _A_ Engine/Core/PlatformMac/MacPlatformIncludes.hhttps://github.com/trillek-team/trillek-client-core/pull/10/files#diff-9(12)
> - _M_ README.mdhttps://github.com/trillek-team/trillek-client-core/pull/10/files#diff-10(73)
> - _M_ src/main.cpphttps://github.com/trillek-team/trillek-client-core/pull/10/files#diff-11(2)
> 
> Patch Links:
> - https://github.com/trillek-team/trillek-client-core/pull/10.patch
> - https://github.com/trillek-team/trillek-client-core/pull/10.diff
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/trillek-team/trillek-client-core/pull/10
> .

DeVaukz commented:
GLFW abstracts windowing systems and GL context setup.  It does not abstract minutia such as how to access the cache prefetch intrinsic (if one exits) or whether the current platform is (or will be built) in 64-bit mode.  That is what is being handled here.  More will be added as necessary.  For example:
- Getting the location where user data should be saved.
- Handling aligned allocations (OS X's allocator is 16-byte aligned by default, other platforms may require a manual implementation).

We're going to need to dip into platform-dependent code and configuration no matter how many libraries we add.  Might as well keep it clean from the start.

adam4813 commented:
Ah. Thanks for the expansion and carry on.
On Apr 20, 2014 10:04 PM, "Devin" notifications@github.com wrote:

> GLFW abstracts windowing systems and GL context setup. It does not
> abstract minutia such as how to access the cache prefetch intrinsic (if one
> exits) or whether the current platform is (or will be built) in 64-bit
> mode. That is what is being handled here. More will be added as necessary.
> For example:
> - Getting the location where user data should be saved.
> - Handling aligned allocations (OS X's allocator is 16-byte aligned by
>   default, other platforms may require a manual implementation).
> 
> We're going to need to dip into platform-dependent code and configuration
> no matter how many libraries we add. Might as well keep it clean from the
> start.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/trillek-team/trillek-client-core/pull/10#issuecomment-40913031
> .

Commit:
Begin Core Infrastructure Feature

Commit:
Add a doxyfile

Commit:
Add some assertion macros.

DeVaukz committed:
Fix the varying names for DEFAULT_TARGETS

adam4813 committed:
Added prerequisites list

adam4813 committed:
Merge branch 'up_develop' into develop

adam4813 committed:
Style fixes in physics

adam4813 committed:
Merge remote-tracking branch 'upstream/master' into develop

adam4813 committed:
Added mouse move and mouse button events

adam4813 committed:
Merge pull request #17 from DeVaukz/develop

Fix the varying names for DEFAULT_TARGETS

DeVaukz committed:
Expand the ReadMe

adam4813 committed:
Merge pull request #20 from DeVaukz/develop

Expand the ReadMe

adam4813 committed:
Exposed transform's entity ID

adam4813 committed:
ResourceSystem now maps a loaders name to its factory

This allows multiple types to share the same type ID (all mesh loaders
could be 1001), but still allow factories for OBJ, MD5Mesh, FBX etc.

adam4813 committed:
Initial OBJ loader code

There vertices need to be stored in a more concise way,
and not just with a separate vector for each component.

adam4813 committed:
Update the test to match the change in the Create parameters

adam4813 committed:
Added some range checking

adam4813 committed:
Transform::SetRotation now creates the correct quaternion

Fixed ResoruceSystemTest and adjusted transform-system-test to match
the new corrected quaternion results.

adam4813 committed:
Added the ability to serialize a specific system

adam4813 committed:
Transforms are now loaded and saved to transforms.json

adam4813 committed:
Merge branch 'develop' into feature-obj-loader

adam4813 committed:
Fixed index checking

The index is 1 based so it must check if it is < or  = to
size and not just <.

adam4813 committed:
Improved OBJ loading by reducing redundancy

adam4813 committed:
Merge branch 'feature-obj-loader' into develop

adam4813 committed:
Updated assets to reflect new OBJ assets

adam4813 committed:
Merge remote-tracking branch 'meisaka/develop' into develop

milesrout commented on README.md:
```diff
@@ -1,2 +1,74 @@
+
+#### Why Add all the directories in each module to the Include Path?
+
+Because we couldn't have duplicate header names anyway.
```
should that be 'because we _shouldn't_ have duplicate header names anyway'? Because you _can_ have duplicate header names.

catageek commented:
Is this PR still relevant ?

adam4813 commented:
I don't think it is.
On Nov 25, 2014 5:07 AM, "catageek" notifications@github.com wrote:

> Is this PR still relevant ?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/trillek-team/trillek-client-core/pull/10#issuecomment-64384440
> .

DeVaukz commented:
Perhaps someday.
